spec.bs

```diff
@@ -1145,38 +1148,45 @@ steps. They return an [=aggregation coordinator=], null or a {{DOMException}}:
 1. Return |origin|.
 
 To <dfn algorithm>obtain the pre-specified report parameters</dfn> given a
-{{SharedStorageRunOperationMethodOptions}} |options|, perform the following
-steps. They return a [=pre-specified report parameters=], null, or a
+{{SharedStorageRunOperationMethodOptions}} |options| and a [=Window/browsing context=] |context|,
+perform the following steps. They return a [=pre-specified report parameters=], null, or a
 {{DOMException}}:
 1. If |options|["{{SharedStorageRunOperationMethodOptions/privateAggregationConfig}}"]
     does not [=map/exist=], return null.
 1. Let |privateAggregationConfig| be
     |options|["{{SharedStorageRunOperationMethodOptions/privateAggregationConfig}}"].
+1. Let |calledFromFencedFrame| be true if |context|'s [=browsing context/fenced frame config
+    instance=] is not null, false otherwise.
 1. Let |contextId| be null.
 1. If |privateAggregationConfig|["{{SharedStoragePrivateAggregationConfig/contextId}}"]
     [=map/exists=], set |contextId| to
     |privateAggregationConfig|["{{SharedStoragePrivateAggregationConfig/contextId}}"].
 1. If |contextId|'s [=string/length=] is greater than 64, return a new
     {{DOMException}} with name "`DataError`".
+1. If |calledFromFencedFrame| is true and |contextId| is not null, return a new {{DOMException}}
+    with name "`DataError`".
 1. Let |filteringIdMaxBytes| be the [=default filtering ID max bytes=].
 1. If |privateAggregationConfig|["{{SharedStoragePrivateAggregationConfig/filteringIdMaxBytes}}"]
     [=map/exists=], set |filteringIdMaxBytes| to
     |privateAggregationConfig|["{{SharedStoragePrivateAggregationConfig/filteringIdMaxBytes}}"].
 1. If |filteringIdMaxBytes| is not [=set/contained=] in the [=valid filtering ID
     max bytes range=], return a new {{DOMException}} with name "`DataError`".
+1. If |calledFromFencedFrame| is true and |filteringIdMaxBytes| is not the [=default filtering ID
+    max bytes=], return a new {{DOMException}} with name "`DataError`".
 1. Return a new [=pre-specified report parameters=] with the items:
     : [=pre-specified report parameters/context ID=]
     :: |contextId|
     : [=pre-specified report parameters/filtering ID max bytes=]
     :: |filteringIdMaxBytes|
 
-The {{WindowSharedStorage}}'s {{WindowSharedStorage/run()}} method steps are
-modified in four ways. First, add the following steps just after step 2 ("If
-{{Worklet/addModule()}} has not yet been called, ..."), renumbering later steps
-as appropriate:
+The {{SharedStorageWorklet}}'s {{SharedStorageWorklet/run()}} method steps are
+modified in four ways. First, add the following steps just after step 4 ("[=Assert=]: window is a
+[=Window=]"), renumbering later steps as appropriate:
 <div algorithm="shared-storage-run-monkey-patch-1">
-3. Let |preSpecifiedParams| be the result of [=obtaining the pre-specified
-    report parameters=] given |options|.
+5. Let |context| be <var ignore>window</var>'s [=Window/browsing context=].
```

Comment:
This part does seem to be less about our integration with shared storage than the shared storage spec itself, but probably it's fine to just monkey patch for now. (I'm hoping to move this all into the shared storage spec soon.)

Reply:
Ack. Will leave as is for now.